sympy/geometry/point.py

```diff
@@ -380,19 +380,23 @@ def are_coplanar(cls, *points):
         points = list(uniq(points))
         return Point.affine_rank(*points) <= 2
 
-    def distance(self, p):
-        """The Euclidean distance from self to point p.
+    def distance(self, other):
+        """The Euclidean distance between self and other .
 
         Parameters
         ==========
 
-        p : Point
+        p : LinearEntity or Point
```

Comment:
There could probably be GeometricEntity. Distance from non-linear geometric objects like a circle might be later implemented.

Reply:
Actually `p` is reference to the point object after line
`s, p = Point._normalize_dimension(self, Point(other))`
It should be `point` only.
`p` does not refer to `other` which can either be a GeometricEntity or LinearEntity.